From 0d6ab4c619b13e8e5e0ca48359afc88579fa67c8 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Mon, 12 Oct 2026 03:04:40 +0200
Subject: [PATCH] avcodec/aac/aacdec_usac: reject SBR ratios other than 2:1

Found during triage/review of the security report GITTfi6VsK4u
(ANT-2026-5KX1BXM6), which does not report this.

Fixes: use of uninitialized memory
Fixes: A-codecs/AAC/xhe-aac/Sintel_24kbps_rap5s.mp4
Use of uninitialized memory Replicated through API with MSAN

(cherry picked from commit 0d6ab4c619b13e8e5e0ca48359afc88579fa67c8)
---
 libavcodec/aac/aacdec_usac.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libavcodec/aac/aacdec_usac.c b/libavcodec/aac/aacdec_usac.c
index 5c6d04226d6..021f817fb8f 100644
--- a/libavcodec/aac/aacdec_usac.c
+++ b/libavcodec/aac/aacdec_usac.c
@@ -588,6 +588,10 @@ int ff_aac_usac_config_decode(AACDecContext *ac, AVCodecContext *avctx,
     }
 
     usac->core_sbr_frame_len_idx = get_bits(gb, 3); /* coreSbrFrameLengthIndex */
+    if (usac->core_sbr_frame_len_idx == 2 || usac->core_sbr_frame_len_idx == 4) {
+        avpriv_report_missing_feature(avctx, "USAC SBR with a ratio other than 2:1");
+        return AVERROR_PATCHWELCOME;
+    }
     m4ac->frame_length_short = usac->core_sbr_frame_len_idx == 0 ||
                                usac->core_sbr_frame_len_idx == 2;
 
-- 
2.53.0

